functions/shiritori/index.js

```diff
@@ -14,116 +14,116 @@
 
 'use strict'
 
-const kuroshiro = require('kuroshiro')
+const Kuroshiro = require('kuroshiro')
+const KuromojiAnalyzer = require('kuroshiro-analyzer-kuromoji')
 const wanakana = require('wanakana')
 
 const smallKanas = 'ぁぃぅぇぉゃゅょ'
-let kuroshiroLoaded = false
 
 // 辞書のローディング。
-exports.loaded = new Promise((resolve, reject) => {
-  kuroshiro.init(function (err) {
-    if (err) {
-      reject(err)
-      return
-    }
-    kuroshiroLoaded = true
-    resolve()
-  })
+const kuroshiro = new Kuroshiro()
+const loaded = kuroshiro.init(new KuromojiAnalyzer())
+
+exports.state = Object.freeze({
+  CONTINUE: 0,
+  LOOSE_N: 1,
+  WIN_N: 2,
+  LOOSE_USED: 3,
+  WIN_USED: 4,
+  LOOSE_CHAIN: 5
 })
 
-// しりとりのルールのチェック。
-exports.check = (word, chain) => {
-  if ((chain === undefined) || (chain.length === 0)) {
-    return true
-  }
+// https://github.com/hexenq/kuroshiro/issues/64
+async function toHiragana (word) {
+  const hira = await kuroshiro.convert(word, { to: 'hiragana' })
+  return wanakana.toHiragana(hira)
+}
 
-  if (kuroshiroLoaded) {
-    word = kuroshiro.toHiragana(word)
-    chain = chain.map((e) => kuroshiro.toHiragana(e))
+// しりとりのルールのチェック。
+exports.check = (word, chain) => loaded.then(async () => {
+  // 「ん」で終わるかどうか？
+  const wordHira = await toHiragana(word)
+  if (wordHira[wordHira.length - 1] === 'ん') {
+    return exports.state.LOOSE_N
   }
 
   // 漢字からひらがなにする。
-  const wordHira = wanakana.toHiragana(word)
-  const chainHira = chain.map(wanakana.toHiragana)
+  if ((chain === undefined) || (chain.length === 0)) {
+    return exports.state.CONTINUE
```

Comment:
I'm not sure that you need `exports` here

Reply:
without `exports` the function won't be able to resolve the symbols since it's not defined as a variable in this scope (or the global scope).

```
  ReferenceError {
    message: 'state is not defined',
  }
```